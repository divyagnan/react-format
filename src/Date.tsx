import * as React from "react";

export interface IDateProps {
  locale?: string;
  options?: {
    localeMatcher?: "lookup" | "best-fit";
    timeZone?: string;
    hour12?: boolean;
    formatMatcher?: "basic" | "best-fit";
    weekday?: "narrow" | "short" | "long";
    era?: "narrow" | "short" | "long";
    year?: "numeric" | "2-digit";
    month?: "numeric" | "2-digit" | "narrow" | "short" | "long";
    day?: "numeric" | "2-digit";
    hour?: "numeric" | "2-digit";
    minute?: "numeric" | "2-digit";
    second?: "numeric" | "2-digit";
    timeZoneName?: "short" | "long";
  };
  children: any;
}

export const Date: React.SFC<IDateProps> = (props): JSX.Element =>
  <span>
    {Intl.DateTimeFormat(props.locale, props.options).format(props.children)}
  </span>;

Date.defaultProps = {
  locale: "en-US",
  options: {}
};
