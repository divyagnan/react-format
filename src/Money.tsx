import * as React from "react";

export interface IMoneyProps {
  locale?: string;
  currency?: string;
  children: any;
}

export const Money: React.SFC<IMoneyProps> = (props): JSX.Element =>
  <span>
    {Intl.NumberFormat(props.locale, {
      style: "currency",
      currency: props.currency
    }).format(props.children)}
  </span>;

Money.defaultProps = {
  locale: "en-US",
  currency: "USD"
};
