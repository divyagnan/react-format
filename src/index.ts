export * from "./Money";
export * from "./Date";
