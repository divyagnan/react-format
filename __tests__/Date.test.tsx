import * as React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Date as DateComponent } from "../src/Date";

const date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));

describe("Date Formatter", () => {
  it("does basic formatting - without config", () => {
    const wrapper = mount(
      <DateComponent>
        {date}
      </DateComponent>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("is responsive to locale changes", () => {
    const wrapper = mount(
      <DateComponent locale="de-DE">
        {date}
      </DateComponent>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("takes and applies an options prop", () => {
    const opts = {};
    const wrapper = mount(
      <DateComponent
        locale="en-US"
        options={{
          timeZone: "America/New_York",
          weekday: "long",
          era: "long",
          year: "2-digit",
          month: "long",
          day: "2-digit",
          hour: "2-digit",
          minute: "2-digit",
          second: "2-digit",
          timeZoneName: "long"
        }}
      >
        {date}
      </DateComponent>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
