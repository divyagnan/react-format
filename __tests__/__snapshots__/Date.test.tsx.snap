// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Formatter does basic formatting - without config 1`] = `
<Component
  locale="en-US"
  options={Object {}}
>
  <span>
    12/19/2012
  </span>
</Component>
`;

exports[`Date Formatter is responsive to locale changes 1`] = `
<Component
  locale="de-DE"
  options={Object {}}
>
  <span>
    19.12.2012
  </span>
</Component>
`;

exports[`Date Formatter takes and applies an options prop 1`] = `
<Component
  locale="en-US"
  options={
    Object {
      "day": "2-digit",
      "era": "long",
      "hour": "2-digit",
      "minute": "2-digit",
      "month": "long",
      "second": "2-digit",
      "timeZone": "America/New_York",
      "timeZoneName": "long",
      "weekday": "long",
      "year": "2-digit",
    }
  }
>
  <span>
    Wednesday, December 19, 12 Anno Domini, 10:00:00 PM Eastern Standard Time
  </span>
</Component>
`;
